src/libraries/System.Private.CoreLib/src/System/Threading/PortableThreadPoolEventSource.cs

```diff
@@ -3,117 +3,286 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using Internal.Runtime.CompilerServices;
 
 namespace System.Threading
 {
-    [EventSource(Name = "Microsoft-Windows-DotNETRuntime", Guid = "{e13c0d23-ccbc-4e12-931b-d9cc2eee27e4}")]
-    public sealed class PortableThreadPoolEventSource : EventSource
+    [EventSource(Name = "Microsoft-Windows-DotNETRuntime", Guid = "e13c0d23-ccbc-4e12-931b-d9cc2eee27e4")]
```

Comment:
Per our conversation offline, you should name this something else, like "Microsoft-DotNETRuntime-PortableThreadPool".  You should also remove the GUID so that EventSource can compute the GUID from the specified name.  This will put you on the supported path, until there is opportunity for the diagnostics folks to work through the details of potentially using NativeRuntimeEventSource.

Reply:
> You should also remove the GUID so that EventSource can compute the GUID from the specified name

We have been using the explicit Guid on all event sources in CoreLib to avoid the expensive reflection to compute it. I do not see a good reason for this one to be on a different plan.

Reply:
Yes, you're right.  @stephentoub and @benaadams mentioned this below.  I had forgotten that this pattern is being used inside the framework.  The GUID just needs to match what EventSource would have generated so that tools don't need to know the GUID, just the name.

Reply:
From our further discussion I have changed the name/guid in mono and undid the EventPipe change in mono, and kept the same name/guid for coreclr and conditioned the EventPipe change for when the portable thread pool is enabled (no change by default). At the moment that seems to be the easiest way to get a similar experience to before in perf profiles.